File: BOJ/Java/2024/25304.java
import java.io.*;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int X = Integer.parseInt(br.readLine());
        int N = Integer.parseInt(br.readLine());

        int totalPrice = 0;
        for (int i = 0; i < N; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            int a = Integer.parseInt(st.nextToken());
            int b = Integer.parseInt(st.nextToken());
            totalPrice += a * b;
        }

        if (X == totalPrice) System.out.println("Yes");
        else System.out.println("No");

    }

}

File: BOJ/C++/2024/25304.cpp
#include <iostream>
using namespace std;

int N, a, b;
long long X, totalPrice;

int main()
{
    cin >> X >> N;
    for (int i = 0; i < N; i++) {
        cin >> a >> b;
        totalPrice += a * b;
    }

    if (X == totalPrice) cout << "Yes" << '\n';
    else cout << "No" << '\n';
    return 0;
}

// 1 000 000 0 (천만) * 0 0 0
